modules/new_profile/jsx/NewProfileIndex.js

```diff
@@ -199,7 +199,7 @@ class NewProfileIndex extends React.Component {
             name = "dobDate"
             label = "Date of Birth"
             minYear = {minYear}
-            maxYear = {maxYear}
+            maxYear = {(new Date()).getFullYear()}
```

Comment:
what if max year is before today though? this maxYear is a meaningful number calculated on the php side using config values (https://github.com/aces/Loris/blob/0910c7bd784e63ab9e0643eb9c0ab52b65627753/modules/new_profile/php/new_profile.class.inc#L169). it doesn't make sense to me project-wise to replace it with today's date.

what's your reasoning for this? i notice that it's going to `22.0-release` branch, is this a bugfix for an issue?

Reply:
@zaliqarosli see the linked related ticket? The issue is to resolve the issue of being able to set a date into the future and doesn't make sense for date of birth.

Reply:
i guess that the restriction on maxYear being today would be if the calculated maxYear from config values is in the future

Reply:
@zaliqarosli yah the maxYear isn't being the current year from the backend. So I thought to add it here on the frontend. thoughts?

Reply:
what we can do is redefine maxYear on line 143 as

```
let today = (new Date()).getFullYear();
let maxYear = (this.state.configData.maxYear > today) ? today : this.state.configData.maxYear
```

Reply:
@zaliqarosli adding it and the current month and current day should still be used. correct? since it solves not having beyond the current date as an option when selecting the date for birthday.

Reply:
^ see the comment above that i just posted haha we're thinking the same thing

Reply:
@zaliqarosli quick question, do you know what the edcDate and edcDateConform is for? Since I'm wondering if that should be able to be set to a greater year than the current date or if it should be similar to the birthday.

Reply:
^ discussed in person